File: samples/cppjs-sample-lib-matrix/playground/Matrix.js
class Matrix extends Array {
    constructor(size, initValue) { super(size); this.fill(initValue); }
    get(i) { return this[i]; }
    multiple(otherMatrix) {
        const size = Math.sqrt(this.length);
        const result = new Matrix(this.length, 0);

        for (let i = 0; i < size; ++i) {
            for (let j = 0; j < size; ++j) {
                for (let k = 0; k < size; ++k) {
                    result[i * size + j] += this[i * size + k] * otherMatrix[k * size + j];
                }
            }
        }
        return result;
    }
}

File: packages/cppjs-sample-lib-prebuilt-matrix/src/native/Matrix.cpp
#include "Matrix.h"

int Matrix::get(int i) { return this->at(i); }

std::shared_ptr<Matrix> Matrix::multiple(std::shared_ptr<Matrix> b) {
    int size = sqrt(this->size());
    auto result = std::make_shared<Matrix>(this->size(), 0);

    for (int i = 0; i < size; i += 1) {
        for (int j = 0; j < size; j += 1) {
            for (int k = 0; k < size; k += 1) {
                (*result)[i * size + j] += this->at(i * size + k) * (*b)[k * size + j];
            }
        }
    }
    return result;
}
